packages/block-editor/src/utils/pre-parse-patterns.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, select } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../store';
+
+export function usePreParsePatterns() {
+	const patterns = useSelect(
+		( _select ) =>
+			_select( blockEditorStore ).getSettings()
+				.__experimentalBlockPatterns,
+		[]
+	);
+
+	useEffect( () => {
+		if ( ! patterns?.length ) {
+			return;
+		}
+
+		let handle;
+		let index = -1;
+
+		const callback = () => {
+			index++;
+			if ( index >= patterns.length ) {
+				return;
+			}
+
+			const marker = `[usePreParsePatterns] process ${ index } ${ patterns[ index ]?.title }`;
+			window.console.time( marker );
+			select( blockEditorStore ).__experimentalGetParsedBlocks(
```

Comment:
I guess the idea here is to parse one pattern at a time, I wonder if we should instead have a selector to fetch a single pattern by name instead of doing this (memoized function called by the global selector and by this script)